### ISSUE TITLE
代码复现问题

### ISSUE THREAD

**Issue by BooleanYu** [qingzhenduyu/ICAL#9]
作者您好，我觉得您的这篇论文非常不错，但在复现的时候缺遇到了一些问题，代码内容我没有进行任何修改，训练我是用一张3090的显卡来跑的，但最终跑出来的结果验证集确是0 我试了几次都是这样 感觉很奇怪 请作者指点，万分感谢！

**Comment by qingzhenduyu** [comment]
把train loss和 validation loss打印出来看看呢

**Comment by BooleanYu** [comment]
这是部分情况 就是感觉loss不下降

Epoch 175:  37%|███▋      | 660/1768 [03:10<05:19,  3.47it/s, loss=8.86, v_num=2, val_fusion_loss=5.290, val_ExpRate=0.000]
Epoch 175:  37%|███▋      | 660/1768 [03:10<05:19,  3.47it/s, loss=8.85, v_num=2, val_fusion_loss=5.290, val_ExpRate=0.000]
Epoch 175:  37%|███▋      | 661/1768 [03:10<05:18,  3.47it/s, loss=8.85, v_num=2, val_fusion_loss=5.290, val_ExpRate=0.000]
Epoch 175:  37%|███▋      | 661/1768 [03:10<05:18,  3.47it/s, loss=8.9, v_num=2, val_fusion_loss=5.290, val_ExpRate=0.000] 
Epoch 175:  37%|███▋      | 662/1768 [03:11<05:18,  3.47it/s, loss=8.9, v_num=2, val_fusion_loss=5.290, val_ExpRate=0.000]
Epoch 175:  37%|███▋      | 662/1768 [03:11<05:18,  3.47it/s, loss=8.91, v_num=2, val_fusion_loss=5.290, val_ExpRate=0.000]
Epoch 175:  38%|███▊      | 663/1768 [03:11<05:18,  3.47it/s, loss=8.91, v_num=2, val_fusion_loss=5.290, val_ExpRate=0.000]
Epoch 175:  38%|███▊      | 663/1768 [03:11<05:18,  3.47it/s, loss=8.87, v_num=2, val_fusion_loss=5.290, val_ExpRate=0.000]

**Comment by BooleanYu** [comment]
我只保留fusion_loss的时候 loss下降正常

![Image](https://github.com/user-attachments/assets/0646e2dd-e55e-4f23-a6f3-dd024c384197)

**Comment by qingzhenduyu** [comment]
这个loss显然是有问题的，另外查看loss建议用tensorboard 来看，用法你可以百度或者问gpt。
至于这个问题，你现在用多大的learning rate和optimizer？

**Comment by BooleanYu** [comment]
如果按照原代码跑 就有问题 Learning rate 和 optimizer 都是默认设置

**Comment by qingzhenduyu** [comment]
训练数据集有修改吗？
我们之前确实没有在单张卡上训练过，我们之前在4张1080ti，2080ti和A800上都验证过效果。

**Comment by BooleanYu** [comment]
代码 数据集都没有进行任何修改 代码直接是github上下载下来的 数据集是下的你们提供的网盘里的数据 我也很奇怪为啥 搞不懂

**Comment by qingzhenduyu** [comment]
如果确实是啥都没修改过的话。我猜测是loss平衡+sgd 优化器的问题，你可以适当调小一下学习率，或者用adadelta之类的优化器，sgd优化器确实偶尔会抽风。
一般来说，换用不同的optimizer对实验效果影响不大

**Comment by BooleanYu** [comment]
好滴 谢谢师兄解答 我来试试看！ 如果方便的话可以加V聊一聊嘛

**Comment by qingzhenduyu** [comment]
> 好滴 谢谢师兄解答 我来试试看！ 如果方便的话可以加V聊一聊嘛
发我邮箱微信号？我加你吧

**Comment by BooleanYu** [comment]
好滴师兄 发送啦

**Comment by qingzhenduyu** [comment]
> 好滴师兄 发送啦
> 

没收到哎。你是不是发错邮箱了

**Comment by HACBOBO** [comment]
啊 要不直接加我吧 15751768973
等会我就删了
